runtime/jcl/common/mgmtgc.c

```diff
@@ -171,64 +170,37 @@ Java_com_ibm_lang_management_internal_ExtendedGarbageCollectorMXBeanImpl_getLast
 	if (NULL == postMaxArray) {
 		goto fail;
 	}
-	poolNamesArray = (jobjectArray)(*env)->NewObjectArray(env, gcInfo->arraySize, stringClass, NULL);
-	if (NULL == poolNamesArray) {
-		goto fail;
-	}
-
-	omrthread_rwmutex_enter_read(mgmt->managementDataLock);
-
-	(*env)->SetLongArrayRegion(env, initialArray, 0, gcInfo->arraySize, (jlong *) &gcInfo->initialSize[0]);
-	if ((*env)->ExceptionCheck(env)) {
-		goto fail2;
-	}
-	(*env)->SetLongArrayRegion(env, preUsedArray, 0, gcInfo->arraySize, (jlong *) &gcInfo->preUsed[0]);
-	if ((*env)->ExceptionCheck(env)) {
-		goto fail2;
-	}
-	(*env)->SetLongArrayRegion(env, preCommittedArray, 0, gcInfo->arraySize, (jlong *) &gcInfo->preCommitted[0]);
-	if ((*env)->ExceptionCheck(env)) {
-		goto fail2;
-	}
-	(*env)->SetLongArrayRegion(env, preMaxArray, 0, gcInfo->arraySize, (jlong *) &gcInfo->preMax[0]);
-	if ((*env)->ExceptionCheck(env)) {
-		goto fail2;
-	}
-	(*env)->SetLongArrayRegion(env, postUsedArray, 0, gcInfo->arraySize, (jlong *) &gcInfo->postUsed[0]);
-	if ((*env)->ExceptionCheck(env)) {
-		goto fail2;
-	}
-	(*env)->SetLongArrayRegion(env, postCommittedArray, 0, gcInfo->arraySize, (jlong *) &gcInfo->postCommitted[0]);
-	if ((*env)->ExceptionCheck(env)) {
-		goto fail2;
-	}
-	(*env)->SetLongArrayRegion(env, postMaxArray, 0, gcInfo->arraySize, (jlong *) &gcInfo->postMax[0]);
-	if ((*env)->ExceptionCheck(env)) {
-		goto fail2;
-	}
 
-	for (idx = 0; idx < gcInfo->arraySize; ++idx) {
-		jstring poolName = NULL;
-		if (mgmt->supportedMemoryPools > idx) {
-			poolName = (*env)->NewStringUTF(env, mgmt->memoryPools[idx].name);
-		} else {
-			poolName = (*env)->NewStringUTF(env, mgmt->nonHeapMemoryPools[idx - mgmt->supportedMemoryPools].name);
+	{
+		jboolean isCopy = JNI_FALSE;
+		jlong* preUsedArrayElems =       (jlong*) (*env)->GetPrimitiveArrayCritical(env, preUsedArray, &isCopy);
```

Comment:
Each of these needs to be NULL checked. Either declare the variables NULL-initialized, then set them, then NULL check each one  on release, or nest the NULL checks, which will get pretty deep.